### PR TITLE
Specify node feature for slurm job

### PR DESCRIPTION
**Pull request by amandarichardsonn**
This PR adds the method `set_node_feature` to srunSettings that accepts a str or list of strs. Users may now specify node constraints for slurm jobs.

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -31,6 +32,9 @@ Description
 
 Detailed Notes
 
+- Users can now specify node features to a job through
```
Comment (al-rigazzi) [inline doc/changelog.rst:35]: I think we should specify "Slurm job"?

File: doc/changelog.rst [hunk 2]
```diff
@@ -31,6 +32,9 @@ Description
...
+  SrunSettings.set_node_feature. The method accepts a string
```
Comment (al-rigazzi) [inline doc/changelog.rst:36]: Should we put SrunSettings.set_node_feature in ticks as `SrunSettings.set_node_feature` or whatever is needed by .rst syntax?

File: smartsim/settings/slurmSettings.py [hunk 3]
```diff
@@ -243,6 +243,21 @@ def set_broadcast(self, dest_path: t.Optional[str] = None) -> None:
         """
         self.run_args["bcast"] = dest_path
 
+    def set_node_feature(self, feature_list: t.Union[str, t.List[str]]) -> None:
+        """Specify the node feature for this job
+
+        This sets ``--constraint``
```
Comment (al-rigazzi) [inline smartsim/settings/slurmSettings.py:249]: Technically, this actually sets `-C`. Maybe either fix docstring or change line 259?

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
A couple of minor details to fix for consistency reasons, mostly

File: doc/changelog.rst (hunk 2 above)
Comment (amandarichardsonn) [reply doc/changelog.rst:36]: yes! good catch

File: smartsim/settings/slurmSettings.py [hunk 4]
```diff
@@ -243,6 +243,21 @@ def set_broadcast(self, dest_path: t.Optional[str] = None) -> None:
         """
         self.run_args["bcast"] = dest_path
 
+    def set_node_feature(self, feature_list: t.Union[str, t.List[str]]) -> None:
+        """Specify the node feature for this job
+
+        This sets ``--C``
```
Comment (al-rigazzi) [inline smartsim/settings/slurmSettings.py:249]: Only `-C`, i.e. with single dash (don't look at me, it's a Unix convention)

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
One super-minor detail to go!

File: smartsim/settings/slurmSettings.py (hunk 4 above)
Comment (amandarichardsonn) [reply smartsim/settings/slurmSettings.py]: oop nice catch

**Review by al-rigazzi — APPROVED** [review]
LGTM, thanks for fixing the last details!

**Pull request merged**